src/js_realm.hpp

```diff
@@ -286,6 +289,18 @@ void Realm<T>::constructor(ContextType ctx, ObjectType this_object, size_t argc,
                 config.schema_version = 0;
             }
             
+            ValueType migration_value = Object::get_property(ctx, object, migration_string);
+            if (!Value::is_undefined(ctx, migration_value)) {
+                FunctionType migration_function = Value::validated_to_function(ctx, migration_value);
```

Comment:
Third argument to `migration_value` should be `"migration"` (or you can use `migration_string` if you can add something to coerce it to a `const char*`) for a nicer error message.